samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -219,6 +220,11 @@ public void run() {
       //initialize JobCoordinator state
       LOG.info("Starting cluster based job coordinator");
 
+      // write the diagnostics metadata file
+      Optional<String> execEnvContainerId = Optional.ofNullable(System.getenv("CONTAINER_ID"));
+      DiagnosticsUtil.writeMetadataFile(new JobConfig(config).getName().get(), new JobConfig(config).getJobId(),
+          "ApplicationMaster", execEnvContainerId, config);
```

Comment:
Extract/reuse constant for string used across multiple files (METRICS_SOURCE_NAME in CPM)